contracts/staking/GoodFundManager.sol

```diff
@@ -193,7 +193,14 @@ contract GoodFundManager is DAOContract {
 		}
 
 		if (_isBlackListed == false) {
-			if (exist == false) activeContracts.push(_stakingAddress);
+			if (exist == false && _rewardsPerBlock != 0) {
+				activeContracts.push(_stakingAddress);
+			} else {
```

Comment:
else if(exists && _rewardsPerBlock==0)
again DRY.
the whole block from if(_isBlacklisted ==false) should be changed to
```
if(exists && (_isBlacklisted || rewards==0) remove
else if(!exists && !(_isBlacklisted || rewards==0) add

Reply:
also we are waisting gas at lines 178-185 they should be moved to the else if(!exists...)

Reply:
Fixed them

Reply:
If we would move it if(!exists...) then we can't update exist ones with new reward values , blockStart or blockEnd